scripts/packageAdBlock.js

```diff
@@ -26,58 +25,55 @@ async function stageFiles (version, outputDir) {
+  if (path.basename(outputDir).length > 32 && resourceJsonPath !== outputResourceJSON) {
     fs.copyFileSync(resourceJsonPath, outputResourceJSON)
   }
 }
 
-const postNextVersionWork = (datFileName, key, publisherProofKey,
+const postNextVersionWork = (componentSubdir, key, publisherProofKey,
   binary, localRun, version) => {
-  const stagingDir = path.join('build', 'ad-block-updater', datFileName)
+  const stagingDir = path.join('build', 'ad-block-updater', componentSubdir)
   const crxOutputDir = path.join('build', 'ad-block-updater')
-  const crxFile = path.join(crxOutputDir, datFileName ? `ad-block-updater-${datFileName}.crx` : 'ad-block-updater.crx')
-  let privateKeyFile = ''
-  if (!localRun) {
-    privateKeyFile = path.join(key, datFileName ? `ad-block-updater-${datFileName}.pem` : 'ad-block-updater.pem')
-  }
+  const crxFile = path.join(crxOutputDir, componentSubdir === 'default' ? 'ad-block-updater.crx' : `ad-block-updater-${componentSubdir}.crx`)
   stageFiles(version, stagingDir).then(() => {
     if (!localRun) {
+      const privateKeyFile = path.join(key, componentSubdir === 'default' ? 'ad-block-updater.pem' : `ad-block-updater-${componentSubdir}.pem`)
       util.generateCRXFile(binary, crxFile, privateKeyFile, publisherProofKey,
         stagingDir)
     }
     console.log(`Generated ${crxFile} with version number ${version}`)
   })
 }
 
-const getOriginalManifest = (datFileName) => {
+const getOriginalManifest = (componentSubdir) => {
   const manifestsDir = path.join('build', 'ad-block-updater')
-  return path.join(manifestsDir, datFileName, 'manifest.json')
+  return path.join(manifestsDir, componentSubdir, 'manifest.json')
 }
 
-const processDATFile = (binary, endpoint, region, keyDir,
-  publisherProofKey, localRun, datFile) => {
-  // we need the last (build/ad-block-updater/<uuid>) folder name
-  const datFileName = path.dirname(datFile).split(path.sep).pop()
-
-  const originalManifest = getOriginalManifest(datFileName)
+const processComponent = (binary, endpoint, region, keyDir,
+  publisherProofKey, localRun, componentSubdir) => {
+  const originalManifest = getOriginalManifest(componentSubdir)
   const parsedManifest = util.parseManifest(originalManifest)
   const id = util.getIDFromBase64PublicKey(parsedManifest.key)
 
   if (!localRun) {
     util.getNextVersion(endpoint, region, id).then((version) => {
-      postNextVersionWork(datFileName, keyDir, publisherProofKey,
+      postNextVersionWork(componentSubdir, keyDir, publisherProofKey,
         binary, localRun, version)
     })
   } else {
-    postNextVersionWork(datFileName, undefined, publisherProofKey,
+    postNextVersionWork(componentSubdir, undefined, publisherProofKey,
       binary, localRun, '1.0.0')
   }
 }
 
-const getDATFileList = () => {
-  return recursive(path.join('build', 'ad-block-updater'))
-    .filter(file => {
-      return (path.extname(file) === '.dat' && !file.includes('test-data'))
```

Comment:
We don't have `test-data` anymore?

Reply:
The original logic recursively looked for `*.dat` files and then backed up to the parent directory. I changed it to just look for subdirectories containing a `manifest.json` file.

So we do, but it's still ignored here because it doesn't have a `manifest.json` file generated for it. Although I'm not actually sure if it's used anywhere.